### PR TITLE
Added implementation for ht_insert()

### PR DESCRIPTION
**Pull request by matthiasbeyer**
No testing yet, no related issue.

### CODE REVIEW

File: src/libreset/ht.c [hunk 1]
```diff
@@ -41,3 +41,15 @@ ht_del(
...
+
+int
+ht_insert(
+    struct ht* ht,
+    void* data,
+    struct r_set_cfg* cfg
+) {
+    rs_hash hash = cfg->hashf(data);
+    size_t i = hash >> (sizeof(hash) - ht->sizeexp);
+    return (NULL != avl_add(&ht->buckets[i].avl, hash, data, cfg));
```
Comment (neithernut) [inline src/libreset/ht.c]: Will inevitably conflict with PR #162.

Comment (matthiasbeyer) [inline src/libreset/ht.c]: You don't have to write this each time, github will notice it for me! :-P

File: src/libreset/ht.c [hunk 2]
```diff
@@ -52,3 +52,15 @@ ht_find(
...
+
+int
+ht_insert(
+    struct ht* ht,
+    void* data,
+    struct r_set_cfg* cfg
+) {
+    rs_hash hash = cfg->hashf(data);
+    size_t i = hash >> (sizeof(hash) - ht->sizeexp);
+    return (NULL != avl_insert(&ht->buckets[i].avl, hash, data, cfg));
```
Comment (neithernut) [inline src/libreset/ht.c]: `avl_insert` doesn't return a pointer anymore.

**Comment by neithernut** [comment]
ACK for merge (given that you don't want to make some parameters const)

**Comment by matthiasbeyer** [comment]
Rebased and added some `const` stuff in another comit. Please review again (after the CI succeeded).

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
Waiting on #195

**Comment by matthiasbeyer** [comment]
Was already ack'ed, so I'll merge now.

**Pull request merged**